braindecode/datasets/bids.py

```diff
@@ -0,0 +1,210 @@
+        This corresponds to the BIDS entity ``desc``. It is used to provide
+        additional information for derivative data, e.g., preprocessed data
+        may be assigned ``description='cleaned'``.
+    suffixes : str | array-like of str | None
+        The filename suffix. This is the entity after the
+        last ``_`` before the extension. E.g., ``'channels'``.
+        The following filename suffix's are accepted:
+        'meg', 'markers', 'eeg', 'ieeg', 'T1w',
+        'participants', 'scans', 'electrodes', 'coordsystem',
+        'channels', 'events', 'headshape', 'digitizer',
+        'beh', 'physio', 'stim'
+    extensions : str | array-like of str | None
+        The extension of the filename. E.g., ``'.json'``.
+        By default, uses the ones accepted by :function:`mne_bids.read_raw_bids`.
+    datatypes : str | array-like of str | None
+        The BIDS data type, e.g., ``'anat'``, ``'func'``, ``'eeg'``, ``'meg'``,
+        ``'ieeg'``.
+    check : bool
+        If ``True``, only returns paths that conform to BIDS. If ``False``
+        (default), the ``.check`` attribute of the returned
+        :class:`mne_bids.BIDSPath` object will be set to ``True`` for paths that
+        do conform to BIDS, and to ``False`` for those that don't.
+    ignore_json : bool
+        If ``True``, ignores json files. Defaults to ``False``.
+    ignore_nosub : bool
+        If ``True``, ignores all files that are not of the form ``root/sub-*``.
+        Defaults to ``False``.
+    preload : bool
+        If True, preload the data. Defaults to False.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+
+    root: Path | str
+    subjects: str | list[str] | None = None
+    sessions: str | list[str] | None = None
+    tasks: str | list[str] | None = None
+    acquisitions: str | list[str] | None = None
+    runs: str | list[str] | None = None
+    processings: str | list[str] | None = None
+    recordings: str | list[str] | None = None
+    spaces: str | list[str] | None = None
+    splits: str | list[str] | None = None
+    descriptions: str | list[str] | None = None
+    suffixes: str | list[str] | None = None
+    extensions: str | list[str] | None = field(
+        default_factory=lambda: [
+            ".con",
+            ".sqd",
+            ".pdf",
+            ".fif",
+            ".ds",
+            ".vhdr",
+            ".set",
+            ".edf",
+            ".bdf",
+            ".EDF",
+            ".snirf",
+            ".cdt",
+            ".mef",
+            ".nwb",
+        ]
+    )
+    datatypes: str | list[str] | None = "eeg"
+    check: bool = False
+    ignore_json: bool = True
+    ignore_nosub: bool = False
+    preload: bool = False
+    n_jobs: int = 1
+
+    def __post_init__(self):
+        from mne_bids import __version__ as mne_bids_version
+
+        assert (
+            mne_bids_version >= "0.16"
+        ), f"mne-bids version must be >= 0.16, got {mne_bids_version}"
```

Comment:
Should be function

Reply:
Maybe it's not even necessary actually to check this. I already specified this in `pyproject.toml`. For MOABB it's different because one version can lead to a silent bug but here the code itself would just fail because of a missing argument. I will remove these lines